File: java/leetcode/n-ary-tree-level-order-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val,List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> result = new ArrayList();
        int depth = 0;
        return findLevels(root, result, depth);
    }

    private List<List<Integer>> findLevels(Node root, List<List<Integer>> result, int depth) {
        if (root == null) { return result; }
        if (result.size() <= depth) {
            List<Integer> newRow = new ArrayList();
            result.add(newRow);
        }
        result.get(depth).add(root.val);
        for (Node child : root.children) {
            result = findLevels(child, result, depth + 1);
        }
        return result;
    }
}

File: python/leetcode/n-ary-tree-level-order-traversal.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: Node
        :rtype: List[List[int]]
        """
        result = []
        depth = 0
        return self.findLevels(root, result, depth)

    def findLevels(self, root, result, depth):
        if root is None:
            return result
        if len(result) <= depth:
            result.append([])
        result[depth].append(root.val)
        for child in root.children:
            result = self.findLevels(child, result, depth + 1)
        return result
